cmd/controller-manager/main.go

```diff
@@ -187,6 +187,7 @@ func watchConfig(cfg *config.Config, stopCh <-chan struct{}) {
 				case <-stopCh:
 					close(sigChan)
 					return
+				default:
```

Comment:
why add the default branch? this may cause there exist multiple configWatchers that notify the eventsCh at the same time?

Reply:
No, this `branch` just used to restart `configWatcher.Watch`

Reply:
I misunderstood the stopCh before, I thought it was used by configWatcher to notify the outer goroutine that it had been finished.
If `configWatcher.Watch` or `eventsCh` first received the msg from stopCh before the outer goroutine, the `configWatcher.Watch` will be restarted forever because case <- stopCh at line 187 won't receive msg again? how to solve this problem?

Reply:
`stopCh` will be closed, so all receiver will receive this event.

Reply:
Got, that's ok

Reply:
Restarting is ok for ErrWatchChannelClosed case, is that ok for other err?